ext/php7/ddtrace.c

```diff
@@ -282,6 +282,28 @@ static zend_object *ddtrace_span_data_clone_obj(zval *old_zv) {
     return new_obj;
 }
 
+#if PHP_VERSION_ID >= 70400
+static zval* ddtrace_span_data_readonly(zval *object, zval *member, zval *value, void **cache_slot) {
+#else
+static void ddtrace_span_data_readonly(zval *object, zval *member, zval *value, void **cache_slot) {
+#endif
+    if (Z_TYPE_P(member) == IS_STRING && zend_string_equals_literal(Z_STR_P(member), "parent")) {
+        zend_throw_exception_ex(zend_ce_exception, 0,
```

Comment:
Is there a reason why we use exception here? For PHP 5, sure, but in general Error is appropriate I'd say?

Reply:
It started out Error, but the test is for all versions, and the only compatible test catches Exception.

Reply:
you could do `} catch (\Exception) { print $error->getMessage() . PHP_EOL; } catch (\Throwable $e) { print $error->getMessage() . PHP_EOL; }` in the test

Reply:
I'd like to have a conversation about this.